docs/troubleshooting/ray-version-compatibility.md

```diff
@@ -0,0 +1,76 @@
+
+In the ray client initialization, there are several checks will be executed in [ray-project/ray/util/client/__init__.py:L115-L137](https://github.com/ray-project/ray/blob/master/python/ray/util/client/__init__.py#L115-L137).
+
+Common cases would be like:
+
+```bash
+...
+
+RuntimeError: Python minor versions differ between client and server: client is 3.8.10, server is 3.7.7
+```
+
+or:
+
+```bash
+...
+RuntimeError: Client Ray installation incompatible with server: client is 2021-05-20, server is 2021-12-07
+```
+
+Some cases may not be so clear:
+
+```bash
+ConnectionAbortedError: Initialization failure from server:
+Traceback (most recent call last):
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/util/client/server/proxier.py", line 622, in Datapath
+    client_id, job_config):
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/util/client/server/proxier.py", line 272, in start_specific_server
+    serialized_runtime_env = job_config.get_serialized_runtime_env()
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/job_config.py", line 99, in get_serialized_runtime_env
+    return self._parsed_runtime_env.serialize()
+AttributeError: 'JobConfig' object has no attribute '_parsed_runtime_env'
+```
+
+```
+Traceback (most recent call last):
+  File "/home/ray/anaconda3/bin/ray", line 8, in <module>
+    sys.exit(main())
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/scripts/scripts.py", line 1938, in main
+    return cli()
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1137, in __call__
+    return self.main(*args, **kwargs)
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1062, in main
+    rv = self.invoke(ctx)
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1668, in invoke
+    return _process_result(sub_ctx.command.invoke(sub_ctx))
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
+    return ctx.invoke(self.callback, **ctx.params)
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 763, in invoke
+    return __callback(*args, **kwargs)
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/autoscaler/_private/cli_logger.py", line 808, in wrapper
+    return f(*args, **kwargs)
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/scripts/scripts.py", line 715, in start
+    services.check_version_info(redis_client)
+  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/services.py", line 823, in check_version_info
+    raise RuntimeError(error_message)
+RuntimeError: Version mismatch: The cluster was started with:
+    Ray: 1.9.0
+    Python: 3.7.7
+This process on node 172.23.110.31 was started with:
+    Ray: 1.10.0
+    Python: 3.7.7
+```
+
+## Solution
+
+In above cases, you need to check if the client ray version is exactly the same with the images version in the ray cluster's configuration. 
```

Comment:
We probably won't need 100% match clients. 
1. Python & Ray minor version matching covers most of the case.
2. upstream ray image has different python versions support. Try to match with your client version. 

![image](https://user-images.githubusercontent.com/4739316/154874876-4028743e-4081-4166-8d82-1e7d48ed41cc.png)

Reply:
yes, after I check,  the version check only goes to minor version, matching minor version is enough and image and python patch version difference is tolerant. I will make some complement before push a amend commit.

Reply:
Yes. It only check minor version level. Great guidance!